tests/base/src/test/java/org/keycloak/test/admin/authz/fgap/UserResourceTypePermissionTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.test.admin.authz.fgap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+import java.util.Set;
+
+import jakarta.ws.rs.core.Response;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.admin.client.resource.ScopePermissionResource;
+import org.keycloak.admin.client.resource.ScopePermissionsResource;
+import org.keycloak.authorization.AdminPermissionsAuthorizationSchema;
+import org.keycloak.common.Profile.Feature;
+import org.keycloak.models.Constants;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.representations.idm.authorization.ScopePermissionRepresentation;
+import org.keycloak.representations.idm.authorization.UserPolicyRepresentation;
+import org.keycloak.test.admin.authz.fgap.UserResourceTypePermissionTest.KeycloakAdminPermissionsServerConfig;
+import org.keycloak.test.framework.annotations.InjectClient;
+import org.keycloak.test.framework.annotations.InjectRealm;
+import org.keycloak.test.framework.annotations.InjectUser;
+import org.keycloak.test.framework.annotations.KeycloakIntegrationTest;
+import org.keycloak.test.framework.realm.ManagedClient;
+import org.keycloak.test.framework.realm.ManagedRealm;
+import org.keycloak.test.framework.realm.ManagedUser;
+import org.keycloak.test.framework.realm.RealmConfig;
+import org.keycloak.test.framework.realm.RealmConfigBuilder;
+import org.keycloak.test.framework.server.KeycloakServerConfig;
+import org.keycloak.test.framework.server.KeycloakServerConfigBuilder;
+
+@KeycloakIntegrationTest(config = KeycloakAdminPermissionsServerConfig.class)
+public class UserResourceTypePermissionTest {
+
+    @InjectRealm(config = RealmAdminPermissionsConfig.class)
+    ManagedRealm realm;
+
+    @InjectClient(ref = Constants.ADMIN_PERMISSIONS_CLIENT_ID, createClient = false)
```

Comment:
`ref` is supposed to be a reference on the client within tests, and not always the same as the `client-id`. We should probably add `String clientId()` to `InjectClient` instead of using `ref`.

Reply:
But I noticed that the `ref` is also used to set the client ID when no clientID is provided in the client configuration/representation. See https://github.com/keycloak/keycloak/blob/main/test-framework/core/src/main/java/org/keycloak/test/framework/injection/SupplierHelpers.java#L40.

I think the same applies to other suppliers such as the realm supplier. Shall we change this in all suppliers then?

Reply:
Created https://github.com/keycloak/keycloak/issues/35995